source/common/upstream/cluster_manager_impl.h

```diff
@@ -516,7 +516,7 @@ class ClusterManagerImpl : public ClusterManager,
       // Current active LB.
       LoadBalancerPtr lb_;
       ClusterInfoConstSharedPtr cluster_info_;
-      Http::AsyncClientImpl http_async_client_;
+      Http::AsyncClientPtr http_async_client_;
```

Comment:
Can you document that this is lazily created, and as such should probably rely on the encapsulation method ?